<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeneratorMainWindow</class>
 <widget class="QMainWindow" name="GeneratorMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1121</width>
    <height>662</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Code Generator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="centralWidgetLayout" stretch="1,4">
    <item>
     <layout class="QVBoxLayout" name="generatedFunctionsVerticalLayout">
      <item>
       <widget class="QTreeWidget" name="generatedFunctionsTreeWidget">
        <property name="uniformRowHeights">
         <bool>true</bool>
        </property>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string notr="true">1</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <spacer name="generatedFunctionsHorizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="removeFunctionButton">
          <property name="text">
           <string>Remove</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="cloneFunctionButton">
          <property name="text">
           <string>Clone</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="newFunctionButton">
          <property name="text">
           <string>New</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="parametersGroupBox">
        <property name="title">
         <string>Parameters</string>
        </property>
        <layout class="QGridLayout" name="parametersGroupBoxLayout" columnstretch="0,0,0,0">
         <item row="3" column="0">
          <widget class="QLabel" name="parameterDataTypeLabel">
           <property name="text">
            <string>Data type</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLineEdit" name="parameterValueEdit"/>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="parameterItemTypeCombo"/>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="parameterNameEdit"/>
         </item>
         <item row="1" column="3" rowspan="4">
          <widget class="QTextBrowser" name="parameterDescriptionEdit">
           <property name="acceptRichText">
            <bool>false</bool>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::LinksAccessibleByKeyboard|Qt::LinksAccessibleByMouse|Qt::TextBrowserInteraction|Qt::TextEditable|Qt::TextEditorInteraction|Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="parameterValueLabel">
           <property name="text">
            <string>Value</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="parameterItemTypeLabel">
           <property name="text">
            <string>Item type</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="parameterNameLabel">
           <property name="text">
            <string>Name</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QLabel" name="descriptionLabel">
           <property name="text">
            <string>Text description / C++ code</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QComboBox" name="parameterDataTypeCombo"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QTreeWidget" name="parameterTreeWidget">
        <property name="columnCount">
         <number>2</number>
        </property>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string notr="true">1</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string notr="true">2</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="itemDeleteButton">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="itemUpButton">
          <property name="text">
           <string>Up</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="itemDownButton">
          <property name="text">
           <string>Down</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="itemNewChildButton">
          <property name="text">
           <string>New Child</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="itemNewSiblingButton">
          <property name="text">
           <string>New Sibling</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1121</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
   </widget>
   <widget class="QMenu" name="menuCode">
    <property name="title">
     <string>Code</string>
    </property>
    <addaction name="actionSet_code_header_h"/>
    <addaction name="actionSet_code_source_cpp"/>
    <addaction name="actionGenerate_code"/>
   </widget>
   <widget class="QMenu" name="menuXFA">
    <property name="title">
     <string>XFA</string>
    </property>
    <addaction name="actionSet_XFA_description"/>
    <addaction name="actionSet_code_header_XFA"/>
    <addaction name="actionSet_code_source_XFA"/>
    <addaction name="actionGenerate_XFA_code"/>
   </widget>
   <widget class="QMenu" name="menuPRC">
    <property name="title">
     <string>PRC</string>
    </property>
    <addaction name="actionSet_PRC_description"/>
    <addaction name="actionSet_code_header_PRC"/>
    <addaction name="actionSet_code_source_PRC"/>
    <addaction name="actionGenerate_PRC_code"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuCode"/>
   <addaction name="menuXFA"/>
   <addaction name="menuPRC"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save As...</string>
   </property>
  </action>
  <action name="actionSet_code_header_h">
   <property name="text">
    <string>Set code header (*.h)</string>
   </property>
  </action>
  <action name="actionSet_code_source_cpp">
   <property name="text">
    <string>Set code source (*.cpp)</string>
   </property>
  </action>
  <action name="actionGenerate_code">
   <property name="text">
    <string>Generate code</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionSet_code_header_XFA">
   <property name="text">
    <string>Set code header XFA</string>
   </property>
  </action>
  <action name="actionSet_code_source_XFA">
   <property name="text">
    <string>Set code source XFA</string>
   </property>
  </action>
  <action name="actionGenerate_XFA_code">
   <property name="text">
    <string>Generate XFA code</string>
   </property>
  </action>
  <action name="actionSet_XFA_description">
   <property name="text">
    <string>Set XFA description</string>
   </property>
  </action>
  <action name="actionSet_code_header_PRC">
   <property name="text">
    <string>Set code header PRC</string>
   </property>
  </action>
  <action name="actionSet_code_source_PRC">
   <property name="text">
    <string>Set code source PRC</string>
   </property>
  </action>
  <action name="actionGenerate_PRC_code">
   <property name="text">
    <string>Generate PRC code</string>
   </property>
  </action>
  <action name="actionSet_PRC_description">
   <property name="text">
    <string>Set PRC description</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
